monai/apps/pathology/transforms/post/array.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence, Union
+
+import numpy as np
+import torch
+
+from monai.config.type_definitions import NdarrayOrTensor
+from monai.transforms.post.array import AsDiscrete, FillHoles, RemoveSmallObjects
+from monai.transforms.transform import Transform
+from monai.utils import TransformBackends, convert_to_numpy, optional_import
+from monai.utils.type_conversion import convert_to_dst_type
+
+label, _ = optional_import("scipy.ndimage.measurements", name="label")
+disk, _ = optional_import("skimage.morphology", name="disk")
+opening, _ = optional_import("skimage.morphology", name="opening")
+watershed, _ = optional_import("skimage.segmentation", name="watershed")
+gaussian, _ = optional_import("skimage.filters", name="gaussian")
+cv2, _ = optional_import("cv2")
+
+__all__ = ["CalcualteInstanceSegmentationMap"]
+
+
+class CalcualteInstanceSegmentationMap(Transform):
+    """
+    Process Nuclei Prediction with XY Coordinate Map.
+
+    Args:
+        threshold_overall: threshold the float values of overall gradient map to int 0 or 1 with specified theashold.
+            Defaults to 0.4.
+        min_size: objects smaller than this size are removed. Defaults to 10.
+        sigma: std. could be a single value, or `spatial_dims` number of values. Defaults to 0.4.
+        kernel_size: the size of the Sobel kernel. Defaults to 21.
+        radius: the radius of the disk-shaped footprint. Defaults to 2.
+
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __init__(
+        self,
+        threshold_overall: float = 0.4,
+        min_size: int = 10,
+        sigma: Union[Sequence[float], float, Sequence[torch.Tensor], torch.Tensor] = 0.4,
+        kernel_size: int = 21,
+        radius: int = 2,
+    ) -> None:
+        self.sigma = sigma
+        self.radius = radius
+        self.kernel_size = kernel_size
+
+        self.remove_small_objects = RemoveSmallObjects(min_size=min_size)
+        self.overall_discreter = AsDiscrete(threshold=threshold_overall)
+        self.fill_holes = FillHoles()
```

Comment:
I don't think this hard-coded set of components can make the transform useful/flexible in general. I think this should be refactored.

Reply:
Thanks for your advice, I just refactored  based on my understanding, could you please help review it again in [this commit](https://github.com/Project-MONAI/MONAI/pull/5173/commits/3bf3bfacd9e269e830e720aa9bcac51dcbacba6d)? Thanks in advance!

Reply:
thanks, this looks good to me now.